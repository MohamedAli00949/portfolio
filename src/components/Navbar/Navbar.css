header.scrolling {
  background-color: #2372afc1;
  backdrop-filter: blur(10px);
  box-shadow: 0px 0px 4px !important;
}

div.nav-container {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  transform: none;
  transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms;
}

button.nav-menu {
  display: none;
}

.logo {
  font-family: "angelika-playfull";
  width: fit-content;
  color: #09c6f9;
  font-size: 50px;
  margin-inline: 20px;
  margin-block: 10px 0;
  text-shadow: -2px -2px #00000059;
}

.nav-section {
  display: flex;
  align-items: center;
  justify-content: center;
  align-content: center;
  flex-wrap: nowrap;
  width: min-content;
}

.nav-section li a {
  color: rgb(255, 255, 255);
  font-size: 20px;
  margin-inline: 15px;
}

.nav-section li a.active {
  text-decoration: none;
  text-shadow: 0px 1px 1px #00000066;
  font-weight: 600;
  color: #3cdbed;
  padding: 10px;
  border-radius: 5px;
  background: #ffffff6b;
  backdrop-filter: blur(10px);
}

.nav-menu {
  background: white;
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: 1000;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.nav-menu ul {
  width: 100%;
  display: flex;
  align-items: center;
  flex-direction: column;
  padding: 20px;
  font-size: 30px;
  justify-content: center;
  gap: 20px;
}

.nav-menu .close {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

/* start responsive design */
@media screen and (max-width: 768px) {
  .nav-section {
    display: none;
  }

  .nav-menu {
    display: flex;
  }

  div.nav-container {
    display: flex;
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
  }

  button.nav-menu {
    display: block;
  }
}

/* end responsive design */
