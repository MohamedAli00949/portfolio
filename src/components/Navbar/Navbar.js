import React, { lazy, useEffect, useState } from "react";
import { AiOutlineMenu, AiOutlineClose } from "react-icons/ai";
import { scrolling, stickyNavbar } from "../../utils/ulits";

const AppBar = lazy(() => import("@mui/material/AppBar"));
const IconButton = lazy(() => import("@mui/material/IconButton"));
const Container = lazy(() => import("@mui/material/Container"));
const Link = lazy(() => import("@mui/material/Link"));
const Slide = lazy(() => import("@mui/material/Slide"));

const Navbar = () => {
  const sections = ["Home", "About", "Portfolio", "Skills", "Contact"];
  const [openMenu, setOpenMenu] = useState(false);

  useEffect(() => {
    window.addEventListener("scroll", stickyNavbar);
    document.addEventListener("scroll", scrolling);
  });

  function checkLink(e) {
    const activeLink = document.querySelector(".section-link.active");

    activeLink.classList.remove("active");

    e.target.classList.add("active");

    setOpenMenu(false);
  }

  if (openMenu) {
    return (
      <div className="nav-menu">
        <div className="close">
          <IconButton
            aria-label="menu"
            style={{
              border: "2px solid #000",
              padding: "10px",
            }}
            onClick={() => {
              setOpenMenu(false);
            }}
          >
            <AiOutlineClose size={30} color="black" />
          </IconButton>
        </div>
        <ul>
          {sections.map((section, index) => (
            <li key={index}>
              <Link
                className={`section-link ${
                  section === "Home" && "active"
                } ${section}`}
                onClick={checkLink}
                href={`#${section}`}
                style={{ textDecoration: "none" }}
              >
                {section}
              </Link>
            </li>
          ))}
        </ul>
      </div>
    );
  }

  return (
    <Container fixed>
      <AppBar
        className="navbar"
        style={{
          width: "100%",
          padding: "5px 20px",
          color: "black",
          flexDirection: "column",
          alignItems: "center",
          justifyContent: "space-between",
          boxShadow: "0 0 0",
          top: "0",
        }}
        color="transparent"
        position="fixed"
        onScroll={(e) => e.target.classList.add("scrolling")}
      >
        <Slide direction="down" in className="nav-container">
          <div>
            <h1 style={{ margin: "0" }}>
              <Link
                className="logo"
                style={{ textDecoration: "none", color: "#09c6f9" }}
                href="/"
              >
                MAF
              </Link>
            </h1>
            <ul className="nav-section">
              {sections.map((section, index) => (
                <li key={index}>
                  <Link
                    className={`section-link ${
                      section === "Home" && "active"
                    } ${section}`}
                    onClick={checkLink}
                    href={`#${section}`}
                    style={{ textDecoration: "none" }}
                  >
                    {section}
                  </Link>
                </li>
              ))}
            </ul>
            <div>
              <IconButton
                aria-label="navMenu"
                className="nav-menu"
                onClick={() => {
                  setOpenMenu(true);
                }}
              >
                <AiOutlineMenu color="white" />
              </IconButton>
            </div>
          </div>
        </Slide>
      </AppBar>
    </Container>
  );
};

export default Navbar;
