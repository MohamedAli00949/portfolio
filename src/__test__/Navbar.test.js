import React, { Suspense } from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import Navbar from "../components/Navbar/Navbar";

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    container.setAttribute("id", "containeer")
    document.body.appendChild(container);
});

afterEach(() => {
    ReactDOM.unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("render footer section", async () => {
    await act(async () => {
        await render(<Suspense><Navbar /></Suspense>)
    })

    container = await document.body;

    expect(container.querySelector("h1 a").textContent).toBe("MAF");

    expect(container.querySelectorAll("ul.nav-section li")[0].querySelector("a").textContent).toBe("Home");
    expect(container.querySelectorAll("ul.nav-section li")[1].querySelector("a").textContent).toBe("About");
    expect(container.querySelectorAll("ul.nav-section li")[2].querySelector("a").textContent).toBe("Portfolio");
    expect(container.querySelectorAll("ul.nav-section li")[3].querySelector("a").textContent).toBe("Skills");
    expect(container.querySelectorAll("ul.nav-section li")[4].querySelector("a").textContent).toBe("Contact");

    expect(container).toMatchSnapshot();
})