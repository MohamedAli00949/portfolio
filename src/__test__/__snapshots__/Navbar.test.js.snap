// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render footer section 1`] = `
<body>
  <div
    id="containeer"
  />
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed navbar mui-fixed css-10srjip-MuiPaper-root-MuiAppBar-root"
        style="width: 100%; padding: 5px 20px; color: black; flex-direction: column; align-items: center; justify-content: space-between; box-shadow: 0 0 0; top: 0px;"
      >
        <div
          class="nav-container"
          style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        >
          <h1
            style="margin: 0px;"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways logo css-1ps4owl-MuiTypography-root-MuiLink-root"
              href="/"
              style="text-decoration: none; color: rgb(9, 198, 249);"
            >
              MAF
            </a>
          </h1>
          <ul
            class="nav-section"
          >
            <li>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link active Home css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="#Home"
                style="text-decoration: none;"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false About css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="#About"
                style="text-decoration: none;"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false Portfolio css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="#Portfolio"
                style="text-decoration: none;"
              >
                Portfolio
              </a>
            </li>
            <li>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false Skills css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="#Skills"
                style="text-decoration: none;"
              >
                Skills
              </a>
            </li>
            <li>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false Contact css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="#Contact"
                style="text-decoration: none;"
              >
                Contact
              </a>
            </li>
          </ul>
          <div>
            <button
              aria-label="navMenu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium nav-menu css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                color="white"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
  </div>
</body>
`;
